### PR TITLE
Do not create "build.properties does not exist" marker on non-PDE #1185

### PR DESCRIPTION
**Pull request by jukzi**
but log a IllegalStateException() to find out how that happens.

https://github.com/eclipse-pde/eclipse.pde/issues/1185

### CODE REVIEW

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/ManifestConsistencyChecker.java [hunk 1]
```diff
@@ -372,13 +375,21 @@ private void validateBuildPropertiesExists(IProject project) {
 			}
 			// if build.properties doesn't exist and build problems != IGNORE, create a marker on the project bug 172451
 			try {
+				if (!project.hasNature(PDE.PLUGIN_NATURE)) {
+					// hod did this happen?
```
Comment (merks) [inline ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/ManifestConsistencyChecker.java:379]: I assume this is temporary so the hod versus how is not so important.

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/ManifestConsistencyChecker.java [hunk 2]
```diff
@@ -364,6 +366,7 @@ private void validateBuildProperties(IProgressMonitor monitor) {
 
 	// Will place a marker on the project if the build.properties does not exist
 	private void validateBuildPropertiesExists(IProject project) {
+
```
Comment (merks) [inline ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/ManifestConsistencyChecker.java:369]: Why add a blank line?

**Review by merks — APPROVED** [review]
It's fine.

**Review by gireeshpunathil — APPROVED** [review]

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/ManifestConsistencyChecker.java (hunk 1 above)
Comment (jukzi) [reply ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/ManifestConsistencyChecker.java]: thanks for mentioning this typos

**Pull request merged**